lucene/core/src/test/org/apache/lucene/index/TestIndexWriterExceptions.java

```diff
@@ -2023,4 +2023,44 @@ public void eval(MockDirectoryWrapper dir) throws IOException {
 
     dir.close();
   }
+
+
+  public void testOnlyRollbackOnceOnException() throws IOException {
+    AtomicBoolean once = new AtomicBoolean(false);
+    InfoStream stream = new InfoStream() {
+      @Override
+      public void message(String component, String message) {
+        if ("TP".equals(component) && "rollback before checkpoint".equals(message)) {
+          if (once.compareAndSet(false, true)) {
+            throw new RuntimeException("boom");
+          } else {
+            throw new AssertionError("has been rolled back twice");
+          }
+
+        }
+      }
+
+      @Override
+      public boolean isEnabled(String component) {
+        return "TP".equals(component);
+      }
+
+      @Override
+      public void close() {
+      }
+    };
+    try (Directory dir = newDirectory()) {
+      try (IndexWriter writer = new IndexWriter(dir, newIndexWriterConfig().setInfoStream(stream)){
+        @Override
+        protected boolean isEnableTestPoints() {
+          return true;
+        }
+      }) {
+      }
+    } catch (RuntimeException x) {
```

Comment:
why `x` ;)

Reply:
yeah why x.... :D

Reply:
I think I have to veto this. I liked x better.

Reply:
let the bikeshedding begin 👯 I call a vote...

Reply:
In the spirit peaceful atmosphere: let's call it e + x = "ex"...

Reply:
I'm really happy there were no votes for runtimeException (IDE generated code fancult vote)